src/service/repo/signedMessageRepo.ts

```diff
@@ -175,9 +179,16 @@ async function createSignedChat(members: string[]) {
   }
 }
 
-async function getSignedChannelDetail(channelType: GetstreamChannelType, channelId: string) {
+async function getSignedChannelDetail(
+  channelType: GetstreamChannelType,
+  channelId: string,
+  timeStamp?: string
+) {
+  const url = timeStamp
+    ? `${baseUrl.getSignedChannelDetail(channelType, channelId)}?last_fetch_date=${timeStamp}`
+    : baseUrl.getSignedChannelDetail(channelType, channelId);
   try {
-    const response = await api.get(baseUrl.getSignedChannelDetail(channelType, channelId));
+    const response = await api.get(url);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a conditional URL construction based on the presence of `timeStamp`. However, there is no validation or sanitization for the `timeStamp` parameter. This could potentially lead to security issues such as injection attacks if the `timeStamp` is not properly validated and sanitized before being used in the URL.

I suggest adding some basic checks to ensure that `timeStamp` is a valid string and does not contain any malicious characters. Here's an example of how you might do this:

```diff
  const url = timeStamp
    ? `${baseUrl.getSignedChannelDetail(channelType, channelId)}?last_fetch_date=${encodeURIComponent(timeStamp)}`
    : baseUrl.getSignedChannelDetail(channelType, channelId);
```

In this change, I've used the `encodeURIComponent` function to encode the `timeStamp` value, which will help prevent potential security issues.


<!-- This is an auto-generated comment by OSS CodeRabbit -->